src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -90,15 +92,16 @@ public TimelineService(final SecuritySettings securitySettings,
         this.transactionTemplate = transactionTemplate;
         this.uuidGenerator = uuidGenerator;
         this.defaultStorage = defaultStorage;
+        this.adminService = adminService;
     }
 
-    public void createTimeline(final String eventTypeName, final String storageId, final Client client)
+    public void createTimeline(final String eventTypeName, final String storageId)
             throws AccessDeniedException, TimelineException, TopicRepositoryException, InconsistentStateException,
             RepositoryProblemException {
         try {
             final EventType eventType = eventTypeCache.getEventType(eventTypeName);
 
-            if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            if (!adminService.isAdmin(AuthorizationService.Operation.WRITE)) {
```

Comment:
You say WRITE, but exception has ADMIN

Reply:
@adyach I know it sound strange, but the reason is that access is forbidden because the user is not an admin (hence the exception), and in particular, he doesn't have the admin READ permission (hence the read). I don't like it very much but I can't find a better alternative.

Reply:
maybe then another exception explain the case ..

Reply:
I could have a AdminPrivilegesRequiredException, with the READ parameter, would that work?

Reply:
do not bother, it is fine as it is